src/controller/java/zap-generated/chip/devicecontroller/ChipStructs.java

```diff
@@ -0,0 +1,2039 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+// THIS FILE IS GENERATED BY ZAP
+package chip.devicecontroller;
+
+import androidx.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Optional;
+
+public class ChipStructs {
+
+  public static class ScenesClusterSceneExtensionFieldSet {
+    public Long clusterId;
+    public Integer length;
+    public Integer value;
+
+    public ScenesClusterSceneExtensionFieldSet(Long clusterId, Integer length, Integer value) {
```

Comment:
Code List `ScenesClusterSceneExtensionFieldSet(1,2,3)` is very hard review and ensure correct ordering of arguments.

Consider having a builder instead since we generate code anyway. This would be easier to read:

```java
ScenesClusterSceneExtensionFieldSet.newBuilder()
    .setClusterId(1)
    .setLength(2)
    .setValue(3)
    .build()
```

Could we use [AutoValue](https://github.com/google/auto/blob/master/value/userguide/index.md) ?

Reply:
Added TODO